force-app/main/default/lwc/templateDesignerCMP/templateDesignerCMP.js

```diff
@@ -876,4 +1087,300 @@ export default class TemplateDesignerCMP extends NavigationMixin(LightningElemen
+    * Once the images drawn in both cases - text & image are saved then their contentversion IDs are captured and using the updateRecord the Watermark_Data__c field on Document_Template__c object is updated based on the selected ID
+    */
+  handleWaterMarkSave() {
+    console.log('handleWaterMarkSave saved >> ');
+    let baseDataLst = [];
+    let canvasText = this.template.querySelector('.canvasText');
+    if (canvasText && this.watermarkText !== '') {
+      let dataURLText = canvasText.toDataURL();
+      baseDataLst.push({ 'text': dataURLText.split(',')[1], title:'Text' });
+    }
+
+    let canvasImage = this.template.querySelector('.canvasImage');
+    if (canvasImage && this.imageUrl) {
+      let dataURLImage = canvasImage.toDataURL();
+      baseDataLst.push({ 'Image': dataURLImage.split(',')[1], title:'Image' });
+    }
+
+    console.log('Canvas Data URL baseDataLst inside handleSave ---- > ', baseDataLst);
+    this.showwatermarkbtn = false;
+    saveContentVersion({ title: "WatermarkImage", base64DataList: baseDataLst, templateId: this.documenttemplaterecordid })
+      .then(result => {
+        console.log('File saved successfully returned ContentVersion ID :', result);
+        this.imageSavedId = result;
+        const fields = {};
+       let watermarkText = (result.filter(obj => Object.keys(obj).some(key => key.includes('Text'))) || [])[0];
+       let watermarkImage = (result.filter(obj => Object.keys(obj).some(key => key.includes('Image'))) || [])[0];
+       watermarkText = watermarkText ? watermarkText[Object.keys(watermarkText)[0]] : null;
+       watermarkImage = watermarkImage? watermarkImage[Object.keys(watermarkImage)[0]] : null;
+
+        const watermarkImageIdText = {
+          name: 'Text',
+          isPrimary: this.checkedValText,
+          contentVersionID: watermarkText,
+          pageOption: this.pageTextOption,
+          fontsize: this.fontSizeValue,
+          opacity: this.opacityValue,
+          color: this.colorValue,
+          rotation: this.rotationValue,
+          textVal: this.watermarkText,
+          pageTextOption: this.pageTextOption
+
+        };
+
+        const watermarkImageIdImage = {
+          name: 'Image',
+          isPrimary: this.checkedValImage,
+          contentVersionID: watermarkImage,
+          pageOption: this.pageImageOption,
+          opacity: this.opacityImageValue,
+          rotation: this.rotationImagevalue,
+          pageImageOption: this.pageImageOption,
+          imageScale: this.imageScalingValue
+        };
+        fields[DOCUMENTTEMPLATEID_FIELD.fieldApiName] = this.documenttemplaterecordid;
+        let jsonDataLst = [watermarkImageIdText, watermarkImageIdImage];
+        fields[WATERMARKDATA_FIELD.fieldApiName] = JSON.stringify(jsonDataLst);
+        const recordInput = { fields };
+        console.log('recordInput to update >> ', recordInput);
+        updateRecord(recordInput)
```

Comment:
Any specific reason to go with ui*recordAPI other than using the existing SaveDocumentTemplate Class methods?